047-TrafficControlWithDapr/Coach/README.md

```diff
@@ -1,26 +1,146 @@
+
+This hack's setup files will create the following resources in your Azure Resource Group. Make sure you can create the following:
+
+- Application Insights
+- Azure Cache for Redis
+- Azure Container Registry
+- Azure Kubernetes Service
+- Event Hub Namespace
+- IoT Hub
+- Key Vault
+- Log Analytics Workspace
+- Logic App (with the Office 365 activity for sending email)
+- Storage Account
+- Service Bus Namespace
+
+_If you can't instantiate some of these resources, you won't be able to complete the part of the challenge that uses them, but you may still be able to complete the other challenges_
+
+#### Special Considerations for Azure Kubernetes Service (AKS)
+
+- AKS requires the ability to create a public IP address. This may be blocked by some organizations. You will either need to get an exception or have an admin create the AKS cluster for you.
+- The `Resources\Infrastructure\bicep\aks.bicep` file specifies the default values for the cluster that will work for this hack. Customize as needed.
+  - 1 Agent Pool with 3 Linux VMs using the **Standard_DS2_v2** SKU.
+  - 3 services using a total of `300m` of CPU & `300Mi` of memory by default, limited to a total of `3000m` of CPU & `600Mi` of memory.
+  - 1 Zipkin service running to monitor communciation between the services.
+- **WARNING:** For simplicity, a Kubernetes secret is used to allow AKS to pull images from the Azure Container Registry via the [admin account](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-authentication?tabs=azure-cli#admin-account). **This is not a best practice**. In a production example, you should use a managed identity & RBAC.
+
+**IMPORTANT:** You will need to register the AKS Dapr extension feature flags in your Azure subscription. Follow the instructions at the link provided below.
+
+[Enable the Azure CLI Extension for Cluster Extensions](https://docs.dapr.io/developing-applications/integrations/azure/azure-kubernetes-service-extension/#enable-the-azure-cli-extension-for-cluster-extensions)
+
+**IMPORTANT:** You will need to register the AKS Workload Identity extension feature flags in your Azure subscription. Follow the instructions at the links provided below.
+
+1.  [Install the AKS Preview Azure CLI Extension](https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#install-the-aks-preview-azure-cli-extension)
+1.  [Register the enableworkloadidentitypreview feature flag](https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#register-the-enableworkloadidentitypreview-feature-flag)
+1.  [Register the enabledoidcissuepreview feature flag](https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#register-the-enableoidcissuerpreview-feature-flag)
```

Comment:
I converted these links to have descriptive text.  However, I don't believe this third links works accurately.  The anchor tag you're linking to doesn't seem to exist in that page.  So the link takes the user to the top of the page.

All 3 of these links are to preview features that are subject to change.  Might want to re-consider this?  Or at least be ready to update as they may change soon (and frequently?)

Reply:
I'll fix the link

As far as the preview feature, there is no GA way to integrate AKS & KeyVault, the Pod Identity feature is also in Preview, the Workload Identity feature (while in Preview) is the direction the AKS PG has indicated is the way this integration will go in the future. I will have to monitor & update these URLs as soon as the AKS PG team makes this feature GA.